src/input/findExistingESLintConfiguration.ts

```diff
@@ -0,0 +1,40 @@
+import { fsFileSystem } from "../adapters/fsFileSystem";
+import { GlobAsync } from "../adapters/globAsync";
+import { ResultStatus, ResultWithDataStatus } from "../types";
+import { ESLintConfiguration } from "./findESLintConfiguration";
+
+export type FindExistingESLintConfigurationDependencies = {
+    globAsync: GlobAsync;
+};
+
+export const findExistingESLintConfiguration = async (
+    dependencies: FindExistingESLintConfigurationDependencies,
+): Promise<ResultWithDataStatus<ESLintConfiguration | string[]>> => {
+    const globResult = await dependencies.globAsync(".eslintrc.{js,json,yaml}");
+    const packageContent = await fsFileSystem.readFile("package.json");
+    if (!(globResult instanceof Error)) {
+        return {
+            data: globResult,
+            status: ResultStatus.Succeeded,
+        };
+    }
+    if (globResult instanceof Error && packageContent instanceof Error) {
+        return {
+            errors: [globResult, packageContent],
+            status: ResultStatus.Failed,
+        };
+    }
+    if (globResult instanceof Error && typeof packageContent === "string") {
```

Comment:
Nit, just marking this down because it confused me at first:

```suggestion
    if (typeof packageContent === "string") {
```